ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+            return 1;
+
+        // Header 3D. Writes it after header
+        if (MMWriteZDescriptionHeaders(hMiraMonLayer, pF, nElements, pZSection))
+            return 1;
+
+        // ZL section
+        pZSection->FlushZL.SizeOfBlockToBeSaved = 0;
+        if (MMAppendBlockToBuffer(&pZSection->FlushZL))
+            return 1;
+
+        if (MMMoveFromFileToFile(pF3d, pF, &pZSection->ZSectionOffset))
+            return 1;
+    }
+
+    if (pF3d)
+        fclose_function(pF3d);
+    if (pszF3d && *pszF3d != '\0')
+        remove_function(pszF3d);
+
+    return 0;
+}
+
+static int MMClosePointLayer(struct MiraMonVectLayerInfo *hMiraMonLayer)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (hMiraMonLayer->ReadOrWrite == MM_WRITTING_MODE)
+    {
+        hMiraMonLayer->nFinalElemCount = hMiraMonLayer->TopHeader.nElemCount;
+        hMiraMonLayer->TopHeader.bIs3d = hMiraMonLayer->bIsReal3d;
+
+        if (MMWriteHeader(hMiraMonLayer->MMPoint.pF, &hMiraMonLayer->TopHeader))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    hMiraMonLayer->MMPoint.pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+        hMiraMonLayer->OffsetCheck = hMiraMonLayer->nHeaderDiskSize;
+
+        // TL Section
+        hMiraMonLayer->MMPoint.FlushTL.SizeOfBlockToBeSaved = 0;
+        if (MMAppendBlockToBuffer(&hMiraMonLayer->MMPoint.FlushTL))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    hMiraMonLayer->MMPoint.pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+        if (MMMoveFromFileToFile(hMiraMonLayer->MMPoint.pFTL,
+                                 hMiraMonLayer->MMPoint.pF,
+                                 &hMiraMonLayer->OffsetCheck))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    hMiraMonLayer->MMPoint.pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+
+        fclose_function(hMiraMonLayer->MMPoint.pFTL);
+        if (*hMiraMonLayer->MMPoint.pszTLName != '\0')
+            remove_function(hMiraMonLayer->MMPoint.pszTLName);
+
+        if (MMClose3DSectionLayer(
+                hMiraMonLayer, hMiraMonLayer->TopHeader.nElemCount,
+                hMiraMonLayer->MMPoint.pF, hMiraMonLayer->MMPoint.pF3d,
+                hMiraMonLayer->MMPoint.psz3DLayerName,
+                &hMiraMonLayer->MMPoint.pZSection, hMiraMonLayer->OffsetCheck))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    hMiraMonLayer->MMPoint.pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+    }
+    if (hMiraMonLayer->MMPoint.pF)
+        fclose_function(hMiraMonLayer->MMPoint.pF);
```

Comment:
```suggestion
    {
        fclose_function(hMiraMonLayer->MMPoint.pF);
        hMiraMonLayer->MMPoint.pF = nullptr;
    }
```

Reply:
Done. Sometimes it seems I am a little amateur.